include/flatbuffers/flatbuffers.h

```diff
@@ -422,9 +422,17 @@ class DefaultAllocator : public Allocator {
     return new uint8_t[size];
   }
 
+  static void *alloc(size_t size) {
```

Comment:
Why do we need these? I don't see them being used anywhere.

Reply:
Good catch. I added it for symmetry. Later on never needed it. If it's really bothersome, I would drop it.

Reply:
A pointer to `DefaultAllocator::dealloc` is passed to `grpc_slice_new` in `MessageBuilder(FlatBufferBuilder &&)`. `DefaultAllocator::alloc` is not used anywhere.

Reply:
I generalized the MessageBuilder move constructor to accept a deallocator. The default is `DefaultAllocator::dealloc`. This change enables conversion of `FlatBufferBuilder` that reserves memory out of a custom allocator with a global state.

Reply:
@aardappel  I zapped the unused `alloc` function from `DefaultAllocator`. Do you think the PR can be merged soonish?